// nylas/lib/esm/resources/drafts.js
import { Messages } from './messages.js';
import { Resource } from './resource.js';
/**
 * Nylas Drafts API
 *
 * The Nylas Drafts API allows you to list, find, update, delete, and send drafts on user accounts.
 */
export class Drafts extends Resource {
    /**
     * Return all Drafts
     * @return A list of drafts
     */
    list({ identifier, queryParams, overrides, }) {
        return super._list({
            queryParams,
            overrides,
            path: `/v3/grants/${identifier}/drafts`,
        });
    }
    /**
     * Return a Draft
     * @return The draft
     */
    find({ identifier, draftId, overrides, }) {
        return super._find({
            path: `/v3/grants/${identifier}/drafts/${draftId}`,
            overrides,
        });
    }
    /**
     * Return a Draft
     * @return The draft
     */
    create({ identifier, requestBody, overrides, }) {
        const form = Messages._buildFormRequest(requestBody);
        return this.apiClient.request({
            method: 'POST',
            path: `/v3/grants/${identifier}/drafts`,
            form,
            overrides,
        });
    }
    /**
     * Update a Draft
     * @return The updated draft
     */
    update({ identifier, draftId, requestBody, overrides, }) {
        const form = Messages._buildFormRequest(requestBody);
        return this.apiClient.request({
            method: 'PUT',
            path: `/v3/grants/${identifier}/drafts/${draftId}`,
            form,
            overrides,
        });
    }
    /**
     * Delete a Draft
     * @return The deleted draft
     */
    destroy({ identifier, draftId, overrides, }) {
        return super._destroy({
            path: `/v3/grants/${identifier}/drafts/${draftId}`,
            overrides,
        });
    }
    /**
     * Send a Draft
     * @return The sent message
     */
    send({ identifier, draftId, overrides, }) {
        return super._create({
            path: `/v3/grants/${identifier}/drafts/${draftId}`,
            requestBody: {},
            overrides,
        });
    }
}
