// nylas/lib/cjs/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.objKeysToSnakeCase = exports.objKeysToCamelCase = exports.createFileRequestBuilder = void 0;
const change_case_1 = require("change-case");
const fs = require("fs");
const path = require("path");
const mime = require("mime-types");
function createFileRequestBuilder(filePath) {
    const stats = fs.statSync(filePath);
    const filename = path.basename(filePath);
    const contentType = mime.lookup(filePath) || 'application/octet-stream';
    const content = fs.createReadStream(filePath);
    return {
        filename,
        contentType,
        content,
        size: stats.size,
    };
}
exports.createFileRequestBuilder = createFileRequestBuilder;
/**
 * A utility function that recursively converts all keys in an object to a given case.
 * @param obj The object to convert
 * @param casingFunction The function to use to convert the keys
 * @param excludeKeys An array of keys to exclude from conversion
 * @returns The converted object
 * @ignore Not for public use.
 */
function convertCase(obj, casingFunction, excludeKeys) {
    const newObj = {};
    for (const key in obj) {
        if (excludeKeys?.includes(key)) {
            newObj[key] = obj[key];
        }
        else if (Array.isArray(obj[key])) {
            newObj[casingFunction(key)] = obj[key].map(item => {
                if (typeof item === 'object') {
                    return convertCase(item, casingFunction);
                }
                else {
                    return item;
                }
            });
        }
        else if (typeof obj[key] === 'object' && obj[key] !== null) {
            newObj[casingFunction(key)] = convertCase(obj[key], casingFunction);
        }
        else {
            newObj[casingFunction(key)] = obj[key];
        }
    }
    return newObj;
}
/**
 * A utility function that recursively converts all keys in an object to camelCase.
 * @param obj The object to convert
 * @param exclude An array of keys to exclude from conversion
 * @returns The converted object
 * @ignore Not for public use.
 */
function objKeysToCamelCase(obj, exclude) {
    return convertCase(obj, change_case_1.camelCase, exclude);
}
exports.objKeysToCamelCase = objKeysToCamelCase;
/**
 * A utility function that recursively converts all keys in an object to snake_case.
 * @param obj The object to convert
 * @param exclude An array of keys to exclude from conversion
 * @returns The converted object
 */
function objKeysToSnakeCase(obj, exclude) {
    return convertCase(obj, change_case_1.snakeCase, exclude);
}
exports.objKeysToSnakeCase = objKeysToSnakeCase;
